docs/dev/framework/caching.md

```diff
@@ -217,24 +217,104 @@ Moreover, you don't have to register to all the different callbacks such as `ons
 You can register to the [`oninvalidate_cache_tags` callback][5] and add your own tags.
 
 
-## Fragments and Edge Side Includes
+## Caching Fragments
 
 In Contao, content elements and front end modules can be implemented as so called
-_fragment controllers_. These fragments are then rendered with their defined renderer 
-and merged into the main content. Each fragment can also provide its own response
-and thus define whether it can be cached or not. If a fragment returns a response
-with a `Cache-Control: private` header for example, then the page on which the fragment
-is visible cannot be cached. On the other hand, if the fragment can be cached, it
-can provide its own cache tags, as mentioned previously.
-
-Contao brings its own `forward` fragment renderer, which provides the fragment with 
-a full clone of the request. This provides the fragment with the full `POST` data
-for example, contrary to Symfony's default `inline` renderer.
-
-The renderer can also be set to `esi`. In that case, if Symfony detects that it 
-is talking to  a gateway cache that supports ESI (like Symfony's built in reverse 
-proxy, that Contao uses), it generates an ESI include tag. See also [Symfony's documentation][esi] 
-on _Edge Side Includes_.
+_fragment controllers_. These fragments are then rendered with their defined renderer
+and merged into the main content. Each fragment returns a response and thus can
+tell whether it can be cached or not.
+
+
+There are two fundamentally different ways to render fragments:
+ 1. **Inline:** Inside the main page request and merged with the content before caching. In Symfony, this is handled
+    by the `inline` renderer, but Contao provides its own default `forward` renderer, which copies the original
+    request (whereas Symfony does not) to e.g. pass POST data to the subrequest.
+ 2. **Subrequest:** By generating an (internal) URL for the fragment, the fragment content is merged **after** caching
```

Comment:
This part, for example, is incorrect. It's not "merged after caching the page". It's rather a placeholder in the original response that is then treated by someone else. In case of ESI it's a reverse proxy. In case of hinclude it's the JS on client-side.

Reply:
Changed in 29eeb9c07ab2b331be81a108d0f16b3a4535af4e